addons/website_slides/wizard/slide_channel_invite.py

```diff
@@ -17,40 +17,40 @@ class SlideChannelInvite(models.TransientModel):
     _name = 'slide.channel.invite'
     _description = 'Channel Invitation Wizard'
 
-    @api.model
-    def _default_email_from(self):
-        if self.env.user.email:
-            return formataddr((self.env.user.name, self.env.user.email))
-        raise UserError(_("Unable to post message, please configure the sender's email address."))
-
-    @api.model
-    def _default_author_id(self):
-        return self.env.user.partner_id
-
     # composer content
-    subject = fields.Char('Subject')
-    body = fields.Html('Contents', default='', sanitize_style=True)
+    subject = fields.Char('Subject', compute='_compute_subject_and_body', readonly=False)
+    body = fields.Html('Contents', default='', sanitize_style=True, compute='_compute_subject_and_body', readonly=False)
     attachment_ids = fields.Many2many('ir.attachment', string='Attachments')
     template_id = fields.Many2one(
         'mail.template', 'Use template',
         domain="[('model', '=', 'slide.channel.partner')]")
     # origin
-    email_from = fields.Char('From', default=_default_email_from)
+    email_from = fields.Char('From', compute='_compute_email_from')
     author_id = fields.Many2one(
         'res.partner', 'Author',
-        ondelete='set null', default=_default_author_id)
+        ondelete='set null', compute='_compute_author_id')
     # recipients
     partner_ids = fields.Many2many('res.partner', string='Recipients')
     # slide channel
     channel_id = fields.Many2one('slide.channel', string='Slide channel', required=True)
     channel_url = fields.Char(related="channel_id.website_url", readonly=True)
 
-    @api.onchange('template_id')
-    def _onchange_template_id(self):
-        """ Make the 'subject' and 'body' field match the selected template_id """
-        if self.template_id:
-            self.subject = self.template_id.subject
-            self.body = self.template_id.body_html
+    @api.depends('template_id')
+    def _compute_subject_and_body(self):
+        for invite in self:
+            if invite.template_id:
+                invite.subject = invite.template_id.subject
+                invite.body = invite.template_id.body_html
+
+    def _compute_email_from(self):
```

Comment:
I'm wondering if we need fields for email_from and author_id at all.

The action_invite code could be updated to:
```py
if not self.env.user.email:
    raise UserError(_("Unable to post message, please configure the sender's email address."))

mail_values = []
for partner_id in self.partner_ids:
    slide_channel_partner = self.channel_id._action_add_members(partner_id)
    if slide_channel_partner:
        mail_values.append(self._prepare_mail_values(slide_channel_partner, self.env.user.email, self.env.user.partner_id))
```

And then add email_from and author_id parameters to the prepare method.

What do you think?

Reply:
It's indeed a good idea I think